Color code relative altitude of FLARM traffic in map window

bomilkar opened the pull request:
<!--

Thank you for your interest in contributing to XCSoar! Please read the
following information to make it easier for us to review your changes.

We appreciate if you make sure to:

  - Document the changes (in the NEWS.txt file, and the manual when relevant)
  - Enable maintainer edits[1] (in case we need to help with the PR)
  - Check your commits and their messages (so they're all tidy and coherent)

[1] https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

-->


Brief summary of the changes
----------------------------

<!--
Please note that the commit messages is where detailed descriptions of the
changes should be made - this section is just for a brief summary/overview.
-->


Related issues and discussions
------------------------------
Addresses Issue #480 
<!--
Please link any relevant issues or forum posts here, for reference.

If this PR resolves an existing issue, please write "Closes #1234" so that
the issue is closed automatically when this PR is merged.
-->

It color codes FLARM traffic symbols on the map (little triangles). Traffic 50m and deeper below is coded green. Traffic 50m and above is coded blue. Traffic in the +/-50m range is coded orange.
A FLARM alert overrides this color coding: the symbol is red (unchanged behavior).

MaxKellermann commented:
This PR contains text which is missing in the commit message.
You're using the `warning_brush` for traffic which is not warned about by the FLARM. I'm not sure if that is a good idea, because that color is reserved for warnings.
I'm not even sure if adding more colors is a good idea at all.

bomilkar commented:
I can think of two ways to signal relative altitude on the map:
1. color code the traffic symbols (little triangles) depending on the relative altitude
2. add numbers to the traffic symbols

I don't think adding more text to the busy map is a good idea.
Hence I vote for option 1.
![ColorCoded_FLARMTraffic_5](https://user-images.githubusercontent.com/20871255/114050317-732b6800-988c-11eb-8a9a-6ce3d24db252.JPG)

MaxKellermann commented:
This PR does two changes in one commit - the green color and the orange color. The latter is questionable because it (ab)uses an existing color for a new meaning. The former is a useful feature, just the color is a matter of taste. But I can't merge one without the other. Do one thing in one commit!

bomilkar commented:
Let me restate the intent because I’m not sure I made myself clear.
I want to identify the FLARM targets in the same altitude range as me (+/- 50m). I’m using color to visualize this as opposed to using text. So I need 2 colors: color1 for in range, color2 for outside of altitude range.
As a cheap by-product I can differentiate _below range_ and _above range_. That splits color2 into two: color2_above and color2_below.

One can debate forever which real colors to pick. I’m suggesting the following:
color1: orange, same color as FLARM Warning color. As far as I can tell `warning_brush` (which uses that `warning_color`) is defined but never really used.
color2_above: blue as the sky, and the same color as it currently uses for FLARM traffic symbols
color2_below: green as the grass. That’s a newly defined color.
Unchanged remains the use of alarm_color.

bomilkar commented:
With the above in mind, I suggest I split this into 2 commits:

1.  identify the FLARM targets in the same altitude range as me.
I can use a different brush and color if we don’t want to (re)use the otherwise unused FLARM Warning color. But then: what color?

2. split the _outside_range_ in two: _below range_ and _above range_.
I guess here I can use green and blue.

Would that be OK, @MaxKellermann ?
